vm/src/pyobject.rs

```diff
@@ -695,10 +695,10 @@ impl Default for PyContext {
 /// This is an actual python object. It consists of a `typ` which is the
 /// python class, and carries some rust payload optionally. This rust
 /// payload can be a rust float or rust int in case of float and int objects.
-pub struct PyObject {
+pub struct PyObject<T: ?Sized> {
```

Comment:
What is the effect of this template on the code size? Did you check this? This is of interest to the wasm build, since it may be intended for web.

Reply:
There aren't really any methods on PyObject, so making it generic won't duplicate much code; it's just a container.

Reply:
We could throw some `#[inline]`s on PyObject::new and PyObject::into_ref but those are the only ones that are really generic.

Reply:
This increases the binary size by about 2.4% for me, which seems acceptable.

(A side note - moving to the new args style should decrease the binary size because there will only be one monomorphization per distinct function signature, whereas `arg_check!` will generate type checking code for each one.)